File: 938 - Range Sum of BST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
let res = 0, l = 0, h = 0;
var rangeSumBST = function(root, low, high) {
    res = 0, l = low, h = high;
    traverse(root);
    return res;
};

function traverse(root) {
    if (root == null) return;
    if (root.val < l) {
        traverse(root.right);
    } else if (root.val > h) {
        traverse(root.left);
        traverse(root.right);
    } else {
        res += root.val;
        traverse(root.left);
        traverse(root.right);
    }
}

File: 938 - Range Sum of BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    res = 0
    low, high = 0, 0
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        self.res = 0
        self.low, self.high = low, high
        self.traverse(root)
        return self.res
    def traverse(self, root):
        if root is None:
            return
        if root.val < self.low:
            self.traverse(root.right)
        elif root.val > self.high:
            self.traverse(root.left)
        else:
            self.res += root.val
            self.traverse(root.left)
            self.traverse(root.right)